plugins/catalog-backend/src/service/router.ts

```diff
@@ -55,10 +59,22 @@ export async function createRouter(
   if (entitiesCatalog) {
     router
       .get('/entities', async (req, res) => {
-        const filter = EntityFilters.ofQuery(req.query);
-        const fieldMapper = translateQueryToFieldMapper(req.query);
-        const entities = await entitiesCatalog.entities(filter);
-        res.status(200).json(entities.map(fieldMapper));
+        const { entities, pageInfo } = await entitiesCatalog.entities({
+          filter: parseEntityFilterParams(req.query),
+          fields: parseEntityTransformParams(req.query),
```

Comment:
Curious what's the conceptual difference and split between filters and fields?

Reply:
`SELECT <filters*> FROM entities WHERE <filters>` essentially :)

Reply:
Filter selects among the entities.

Fields selects among the fields of the json structure of each returned entity. For example if you only want to return

```json
[
  { "metadata": { "uid": "a" }},
  { "metadata": { "uid": "b" }},
  { "metadata": { "uid": "c" }}
]
```

and exclude everything else to make things faster and use less bandwidth

Reply:
Nice. I guess the naming threw me off because `filters` would have fields as well. Although using `select` here is to SQL like.